vrandall66/NASA-Pod-Vue
# Implement loading gif, add photo/:date path and successfully pass photo object to new path!

## What does this PR do?
- [x] Implements loading gif on `MonthContainer.vue`
- [x] Adds route `/photo/:date` to `router.js`
- [x] Add `PreviousPhoto.vue` and `PreviousPhotoDetail.vue`
- [x] Moves navigation into `NavBar.vue`

### Where should the reviewer start?
- [ ] In `router.js` on line 27, notice the `props: true`
- [ ] In `PhotosOfTheMonth.vue` on lines 4, 21, 36 and 37, notice the setLoading() logic and how it is being passed to `MonthContainer.vue`
- [ ] In `MonthContainer.vue` on lines 4, 5, and 9 you will see the `v-if` and `v-bind:"require()"` needed to conditionally render the loading gif versus the images
- [ ] In `MonthContainer.vue` on lines 16, and 31 through 36, you will see how I had to pass props/params to the previous route (A.K.A `/photo/:id`)
- [ ] Please look over `PreviousPhotoDetail.vue`
### How should this be manually tested?
Pull down and click on a photo from the "Photos from this month" page! Check out the details!
### Have tests been written for all functionality?
No.
### Any background context you want to provide?
- It took me forever to figure out how to pass props/params utilizing router. I need to double check whether it is the params or the props that is actually passing the photo and date information. I know params needs the date for the route, but I went back and forth on this for an embarrassingly long time and will clean it up in the morning.
- I would also like to refactor `PhotoDetail.vue` and `PreviousPhotoDetail.vue` to be more dynamic so there is one reusable component instead of two identical components. Will revisit in the morning.
### What are the relevant tickets?
Closes #11 
### Screenshots (if appropriate)
![2020-01-14 23 28 23](https://user-images.githubusercontent.com/31895658/72410495-ad5b9780-3725-11ea-966f-dd76741a7717.gif)

### Additional Notes
Please refer to the Background context section.
